lib/chef/resource/execute.rb

```diff
@@ -165,7 +165,7 @@ def after_created
       end
 
       def validate_identity_platform(specified_user, password = nil, specified_domain = nil)
-        if Chef::Platform.windows?
+        if node[:os] == "windows"
```

Comment:
You should use `node[:platform_family] == "windows"` here because it may be more compatible with ecosystem tools.

https://github.com/chef/chef/pull/5764#issuecomment-276470185